src/nvme/fabrics.c

```diff
@@ -827,18 +827,12 @@ int nvmf_get_discovery_log(nvme_ctrl_t c, struct nvmf_discovery_log **logp,
 				 name, retries, max_retries, errno);
 			goto out_free_log;
 		}
-
+		if (genctr == le64_to_cpu(log->genctr))
+			break;
 		genctr = le64_to_cpu(log->genctr);
 		nvme_msg(r, LOG_DEBUG,
 			 "%s: discover genctr %" PRIu64 ", retry\n",
 			 name, genctr);
-		ret = nvme_discovery_log(nvme_ctrl_get_fd(c), hdr, log, true);
```

Comment:
But what is updating the `genctr` if has changed?

Reply:
Hmm. Yes, slightly more massaging is required here.